### PR TITLE
Add caller_name argument to TChannel.call and friends

### LINKED ISSUE #356: Override caller name on a per-request basis
This is useful for forwarding use cases where you want to pretend to be the calling service.

Proposal: Add `caller_name` override for transport headers to `TChannel.call` and the corresponding methods on the arg schemes.

### PR DESCRIPTION
**Pull request by abhinav**
Resolves #356.

This will be part of the mash release.

### CODE REVIEW

**Comment by abhinav** [comment]
@blampe @breerly @junchaowu

File: CHANGES.rst [hunk 1]
```diff
@@ -17,6 +17,10 @@ Changes by Version
 - Limit the size of arg1 to 16KB.
 - Fix bug which prevented requests from being retried if the candidate
   connection was previously terminated.
+- You can now specify an override for a request's ``cn`` header using the
```
Comment (junchaowu) [inline CHANGES.rst]: change the header to transport header. header will confuse.

Comment (abhinav) [inline CHANGES.rst]: word

**Comment by abhinav** [comment]
Putting this on hold. We may be able to solve the problem without exposing this API.

CC @junchaowu

**Comment by junchaowu** [comment]
lgtm

**Comment by breerly** [comment]
Suggest `caller` instead to match YARPC (although I don't feel too strongly)

**Comment by abhinav** [comment]
This PR has been open a while. I don't remember exactly why we were adding this or why it was put on hold.

**Comment by breerly** [comment]
@abhinav presumably someone was writing a tchannel proxy?

**Pull request closed without merging**